### DISCUSSION TITLE
Luma Map Transitions

### DISCUSSION THREAD

**Discussion by decryptedchaos** [olive-editor/olive#2278] [Ideas]
So I've been following Olive for a while now..  and I believe it has great potential.   I recently read the announcement about how Olive is falling behind and becoming stagnate..  I hope it finds a path forward,  I don't have the C++ skill to help sadly..   

But I do have a suggestion,   perhaps even a whole design ideology,   One of the areas I feel Olive is lacking compared to both other open source and commercial NLEs is transitions, (to be fair, in contrast, olive has one of the best chroma keyers I've used so pick and choose)  currently there are only very very basic partially working options, I skimmed through the code and as far as I can tell the two transitions are hard coded.. 

I feel like it should make it much easier to create maintain and process transitions if you just create a function to process a folder of luma map transitions,  but they should absolutely be presented in a add menu or list..   to cite an example  shotcut makes you browse your filesystem to select the png file..  this is such a workflow breaking step it isn't funny..  so i don't think we should ever have too leave the interface to add them..  

And then focus on the compositor,  because that is pretty much where i see most NLEs going..    I could probably come up with a lot more if I put thought into it..   but the desire/vision of the team/community will dictate if that's worth the time..

**Comment by IcedQuinn** [comment]
This _should_ be easy to do with compositor nodes but I'm not sure if we have the right ones yet. These maps are mostly greyscale images and the math is to show source A at every pixel below some cutoff value and B otherwise.
